### PR TITLE
Move PlaidML plugin into this repo

### PR DESCRIPTION
**Pull request by tzerrell**
Move the PlaidML OpenVINO plugin code from the PlaidML repository into this repository.

This also moves to be based on the 2020.4 version of the code.

### CODE REVIEW

**Comment by tzerrell** [comment]
Changed the base to make the actual changes of this PR clearer

**Comment by tzerrell** [comment]
This CMake build is now working with both the single layer tests and benchmark_app. I spent a while trying to get the single layer test skipping file to work, but ultimately found it was not working for me even on other plugins (i.e. the MKLDNN plugin), so I decided it wasn't a priority and just removed tests with known failures. Getting more thorough testing coverage with this in-repo plugin will involve re-adding and fixing tests, but would also benefit from moving forward from 2020.4

**Comment by tzerrell** [comment]
There is some room for CMake upgrades, i.e. replacing `PLAIDML_SRC_PATH` by using `plaidml-config.cmake`

**Comment by tzerrell** [comment]
I don't actually need everyone's review (but certainly I don't object to reviews from people who want to give them!), but I'm tagging people who helped with this since I'm not sure you'd see that this PR was happening otherwise. Thanks for the help!

**Review by flaub — COMMENTED** [review]
LGTM

File: inference-engine/src/inference_engine/ie_plugin_dispatcher.cpp [hunk 1]
```diff
@@ -36,7 +36,8 @@ std::string getPluginName(const std::string& deviceName) {
     static std::map<std::string, std::string> plugunFromDeviceMap = {
         {"CPU", "MKLDNNPlugin"},    {"GPU", "clDNNPlugin"},         {"FPGA", "dliaPlugin"},
         {"MYRIAD", "myriadPlugin"}, {"HDDL", "HDDLPlugin"},         {"GNA", "GNAPlugin"},
-        {"HETERO", "HeteroPlugin"}, {"MULTI", "MultiDevicePlugin"}, {"KMB", "kmbPlugin"}};
+        {"HETERO", "HeteroPlugin"}, {"MULTI", "MultiDevicePlugin"}, {"KMB", "kmbPlugin"},
+        {"PlaidML", "PlaidMLPlugin"}};
```
Comment (brianretford) [inline inference-engine/src/inference_engine/ie_plugin_dispatcher.cpp:40]: every other one is ALLCAPS may PLAIDML or PML?

Comment (flaub) [reply inference-engine/src/inference_engine/ie_plugin_dispatcher.cpp:40]: I wonder what a `plugun` is :)

File: inference-engine/src/plaidml_plugin/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+file(GLOB HEADERS
+        ${CMAKE_CURRENT_SOURCE_DIR}/*.hpp
+)
+
+addVersionDefines(plaidml_plugin.cpp CI_BUILD_NUMBER PLAIDML_VERSION)
+
+add_library(plaidml SHARED IMPORTED)
+
+set_target_properties(plaidml PROPERTIES IMPORTED_LOCATION ${PLAIDML_SRC_PATH}/bazel-bin/plaidml/libplaidml.so)
```
Comment (brianretford) [inline inference-engine/src/plaidml_plugin/CMakeLists.txt:21]: rename PLAIDML_SRC_DIR and make a new property PLAIDML_SO_PATH that defaults to this - I can do this in a follow up too

**Review by brianretford — APPROVED** [review]
Generally LGTM

File: inference-engine/src/plaidml_plugin/CMakeLists.txt (hunk 2 above)
Comment (flaub) [reply inference-engine/src/plaidml_plugin/CMakeLists.txt:21]: Would be good to make use of: https://github.com/plaidml/plaidml/blob/plaidml-v1/plaidml/plaidml-config.cmake

Or basically tweak both sides so they are compatible with each other.

**Pull request merged**